### ISSUE TITLE
Topaz Video Enhance AI performance is abysmal with the new driver.

### ISSUE THREAD

**Issue by gargamel314** [IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT#258]
### Checklist [README]

- [X] Device is not undervolted nor overclocked
- [X] Device is using the [latest drivers](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/wiki/Intel-driver-updates-and-changelogs)
- [X] Application is not cracked, modded and use the latest patch

### Application [Required]

Topaz Video Enhance AI

### Processor / Processor Number [Required]

i7-13700K

### Graphic Card [Required]

Arc A770 LE

### GPU Driver Version [Required]

- [X] 31.0.101.4146
- [ ] 31.0.101.4125
- [ ] 31.0.101.4123
- [ ] 31.0.101.4091
- [ ] 31.0.101.4090
- [ ] 31.0.101.4034
- [ ] 31.0.101.4032
- [ ] 31.0.101.3975
- [ ] 31.0.101.3959
- [ ] 31.0.101.3802
- [ ] 31.0.101.3793
- [ ] 31.0.101.3491
- [ ] 31.0.101.3490

### Rendering API [Required]

- [ ] Vulkan
- [ ] OpenGL
- [ ] DirectX12
- [ ] DirectX11
- [ ] DirectX10
- [ ] DirectX9
- [X] Not applicable

### Windows Build Number [Required]

- [X] Windows 11 22H2
- [ ] Windows 11 21H2
- [ ] Windows 10 22H2
- [ ] Windows 10 21H2
- [ ] Other (Please specify)

### Intel System Support Utility report

[SSU.txt](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/files/10867153/SSU.txt)

### Description and steps to reproduce [Required]

Topaz VEAI rendering using driver 4146 causes much slower renders, by upwards of 8 times as slow. 

Upscaling a 480p video to 1080p, and interpolating from 23.976fps to 60fps using Proteus & Apollo models.

4125 gave me .08 seconds per frame.

4046 gave me .75 seconds per frame.

Reinstalling 4125 restored the .08 seconds per frame render speed.

Steps to reproduce:
1. Install Driver version 4146.
2. Open Topaz VEAI, and load a 480p movie.
3. Select output resolution 1920x1080p
4. Select output frame rate at 60FPS
5. Select models Apollo (for interpolation) and Proteus manual for upscaling.
6. Click "Preview" or "Export"

Installing the 4125 drivers and following the above steps yielded a much faster render. 

### Device / Platform

ASUS Strix Z690-A D4

### Application / Windows logs

No crashes

**Comment by Arturo-Intel** [comment]
Hello @gargamel314 !

Thank you for reaching us, we will start working on replicating this behavior.
I am just wondering, did you did a clean install of the latest driver (4146)? (using [DDU](https://www.intel.com/content/www/us/en/support/articles/000091878/graphics.html)).

Also, the software you use is this? https://www.topazlabs.com/topaz-video-ai

Thank you for your feedback

**Comment by gargamel314** [comment]
That is the correct software. I did not use DDU before installing 4146. I will try that this evening and report back.

**Comment by gargamel314** [comment]
Uninstalling 4125 using DDU and installing 4146 had the same effect. current frame time is .93spf, it should be .08 spf. Yes, you do have the right software listed. TopazLabs Video Enhance AI version 3.1.7.

**Comment by gargamel314** [comment]
i would also like to add that under 4125, CPU utilization went to 100% while loading the Proteus Model, and then held at 35% once the rendering started. Rendering was at 80%. Under 4146, CPU never really went above 25% and Rendering stayed between 2% and 20%. I've swapped between 4125 and 4146. Every time I do, I get these respective behaviors. A bug report was also submitted at Topaz Labs forums.

**Comment by gargamel314** [comment]
I'm just leaving this post here that contains information about what may be going on:

https://community.topazlabs.com/t/very-slow-performance-on-topaz-photo-ai-and-video-enhance-ai-on-windows-with-intel-gpu/40973/4

**Comment by Arturo-Intel** [comment]
@gargamel314 so with the 4125 you got back the .08spf you were expecting?
![image](https://user-images.githubusercontent.com/92489452/224119100-3edce3c8-2451-494f-b352-0f6421dd59c4.png)

The issues is not reproducible anymore?

**Comment by Arturo-Intel** [comment]
A little more info:

hmm it looks you need to specify the GPU you want to use.
In my config I have a iGPU (device 0) and the Arc770 (device 1)

The "AI Processor" option in preferences used to select what device to use (in Topaz) seems not to work, 
I had to disable the iGPU to force the usage of the ARC. Still the CPU still over 95% of usage.
![image](https://user-images.githubusercontent.com/92489452/224123338-8899b01c-d953-4f7d-943c-2c4d604b0131.png)

Maybe Topaz is always using the  "Auto" option on the AI processor thats why the CPU is always almost 100% and the Arc under the 25%

Based on the link you share, looks like Topaz doesn't know which device to use, and goes for the first it found....

**Comment by gargamel314** [comment]
Hi there,

Yes, that seems like what's happening - it wanted to use the iGPU or CPU instead of Arc. 

It seems that the problem has to do with the OpenVINO runtime used in Topaz, it must be an old version. Someone else figured out if you download the OpenVINO Toolkit and copy the files in the runtime/bin/intel64/Release folder into Topaz VEAI's main directory, it uses the correct device. Only now, I seem to get "unknown errors" that cancel every render in the latest version of VEAI (3.1.8), so I rolled back to 3.1.7. Arc users would have to copy this folder every time the program installs a new version of itself. VEAI is kind of a mess right now for Arc A770!

**Comment by Arturo-Intel** [comment]
I see

We can say that there is a problem with that old version of OpenVINO + Arc drivers, right?
Do you think Topaz will update OpenVINO in the future? 

BTW How often do  you see this "unknown errors"?

**Comment by gargamel314** [comment]
Topaz is aware of the issue, I think they probably will. 

After applying the OpenVINO fix, about 10 minutes into the render, the screen freezes, goes black, and then a second later everything comes back and the render has failed with an unknown error. This happens every time I attempt to upscale a file in Topaz VEAI. Later, after a render had failed, just sitting with VEAI open, the screen continued blacking out, even after I closed off Topaz. I haven't tried restarting it yet. 

I also just went on a bug bash this morning with an unrelated bug involving the EA launcher app - not its games, but its launcher - I would open the app and then just clicking on a game's page in the app would cause a BSOD with a stopcode: DRIVER_VERIFIER_DMA_VIOLATION

This happened once before and fixing it involved physically uninstalling the A770 and all of its software, running DDU, uninstalling the EAP launcher app via Revo Uninstaller, purging every occurrence of Electronic Arts from the registry and then reinstalling everything from scratch. I did this this morning, and I am still dealing with this same issue with Topaz. on the plus side my computer no longer hangs with a blank screen for 12 seconds after POST before Windows loads.

**Comment by Arturo-Intel** [comment]
I see, even with the OpenVINO workaround the problem with the Enhance AI performance is still present?

**Comment by gargamel314** [comment]
The blackout screen is a _different_ problem. I just upgraded to VEAI 3.1.9, they seem to have fixed the problem with OpenVINO and the bad performance. I upgraded without having to overwrite any files. Performance seems to be back to its original state.

The new problem arised with driver 4146. Would it make sense to make a new thread? This problem is that when you run Topaz Video Enhance AI, usually after around 5 or 10 minutes the screen will suddenly freeze, black out for about 3 seconds, and then come back. When everything comes back, VEAI shows the render failed with an "unknown error."  I use two monitors: #1 is a Viewsonic 27" VX2758-2KP-MHD 2560x1440 @ 144Hz. Adaptive sync is enabled. The second is a Dell U2715H 2560x1440 @ 60Hz, it's not Adaptive-sync enabled. Both screens freeze and go black when this happens. Even after closing VEAI, I have noticed the screens black out on Windows Desktop. I've tried the Viewsonic at different refresh rates, even at 59Hz and 60 Hz. his never happens during any kind of gaming app, they remarkably run just fine. 

Things I've tried:
BIOS Settings - I had ASPM enabled w/ L1 enabled to control the idle power, I tried  disabling and nothing changed.
Switching from DP cable to HDMI - no change
DDU - I've used it every time I've changed drivers.
Running at completely stock speeds - no overclock on the GPU or CPU. CPU is slightly undervolted for thermals.
Rolling back VEAI by completely uninstalling using REVO Uninstaller and reinstalling from scratch - same behavior on versions 3.1.6, 3.1.7, 3.1.8, 3.1.9. 
**Rolling back GPU driver. This had success. Everything is running fine on Arc Driver version 4125. This behavior black out screen comes with 4146 and now 4148, however I have seen the screen black out in 4125 when not running Topaz VEAI, but not nearly as frequently. Currently running a render that has been running for 90 minutes in driver version 4125, no problems.**
